tools/packager/main.go

```diff
@@ -112,6 +112,45 @@ func doPackage() error {
 		return errors.Wrap(err, "Failed to write dir to archive")
 	}
 
+	err = tw.Close()
+	if err != nil {
+		return errors.Wrap(err, "Failed to close tar writer")
+	}
+
+	err = zw.Close()
+	if err != nil {
+		return errors.Wrap(err, "Failed to close gzip writer")
+	}
+
+	err = f.Close()
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to close -archivePath %s", archivePath))
+	}
+
+	f, err = os.OpenFile(archivePath, os.O_RDONLY, 0777)
```

Comment:
Could we calculate the sha as we're writing the file, so that it's faster?

Reply:
Thank you for your suggestion.
I have made this change in [this commit](https://github.com/buildpacks/lifecycle/pull/679/commits/f64a1f7c20c52f6fa88ca11b5dcd0ed108155b0a).
I don't think that the performance really matters in this case because in both cases the calculation was very fast, but it appears that this is the convention.